pypsa/inverter_functions.py

```diff
@@ -0,0 +1,512 @@
+"""importing important libraries."""
+import logging
+import numpy as np
+from copy import deepcopy
+logger = logging.getLogger(__name__)
+
+
+def fixed_cosphi(p_input, now, c_attrs, c_list_name):
+    """
+    fix power factor inverter controller.
+    reference : https://www.academia.edu/24772355/
```

Comment:
this link is not accessible without a google or facebook account? Is this on arxiv or open access somewhere?

Generally, I would like it if the documentation and docstrings were self-contained with academic papers listed as additional references.

Can you explain the functionality in a few sentences directly in the docstring?